### PR TITLE
Added a benchmarks directory

### PR DESCRIPTION
**Pull request by jalextowle**
## Description

This PR introduces a benchmarking directory and a benchmark script to calculate the maximum size of a specific type of NFT order (the type used by TokenTrove). 

## Testing instructions
To run these benchmarks accurately, the contracts need to be recompiled using the `constantinople` EIP schedules. Additionally, the web3 provider will also need to be configured to use `constantinople`. 

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/integrations/test/benchmarks/exchange_benchmarks.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    });
+
+    interface MapEIP721BenchmarkResult {
+        gasUsed: number;
+        orderSize: number;
+    }
+
+    async function mapEIP721BinarySearchAsync(start: number, end: number): Promise<MapEIP721BenchmarkResult> {
+        const orderSize = Math.round((start + end) / 2);
+        console.log(`Testsing ${orderSize}`);
```
Comment (mzhu25) [inline contracts/integrations/test/benchmarks/exchange_benchmarks.ts]: 🇨🇦 
```suggestion
        console.log(`Testing ${orderSize}`);
```

File: contracts/integrations/test/benchmarks/exchange_benchmarks.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            gasUsed = tx.gasUsed;
+        } catch (err) {
+            error = err;
+        }
+
+        if (error !== undefined) {
+            if (orderSize === start) {
+                return { gasUsed, orderSize };
+            }
+            console.log(`Failure, searching [${start}, ${orderSize - 1}]`); // tslint:disable-line:no-console
```
Comment (mzhu25) [inline contracts/integrations/test/benchmarks/exchange_benchmarks.ts]:
```suggestion
            console.log(`Failure, searching [${start}, ${orderSize - 1}]`);
```

File: contracts/integrations/test/benchmarks/exchange_benchmarks.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        taker = new Taker({ name: 'Taker', deployment });
+        await taker.configureERC20TokenAsync(takerToken);
+        await taker.configureERC20TokenAsync(deployment.tokens.weth, deployment.staking.stakingProxy.address);
+    });
+
+    it(`MAP EIP721 Benchmark -- max gas used [${START}/${END}]`, async () => {
+        console.log(START);
+        console.log(END);
+        const result = await mapEIP721BinarySearchAsync(START, END);
+        console.log(result); // tslint:disable-line:no-console
```
Comment (mzhu25) [inline contracts/integrations/test/benchmarks/exchange_benchmarks.ts]:
```suggestion
        console.log(result);
```

File: contracts/integrations/test/benchmarks/exchange_benchmarks.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import { DummyERC721TokenContract } from '@0x/contracts-erc721';
```
Comment (hysz) [inline contracts/integrations/test/benchmarks/exchange_benchmarks.ts]: Nit - "exchange_benchmarks" is pretty broad. In the interest of keeping these files small, perhaps something more specific / modular?

**Pull request closed without merging**